testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py

```diff
@@ -265,10 +335,182 @@ def test_peer_2_peer_sanity_push(setup_p2p_test):
+
+
+@pytest.mark.sanity
+@pytest.mark.listener
+@pytest.mark.replication
+@pytest.mark.p2p
+@pytest.mark.changes
+def test_peer_2_peer_sanity_push_one_shot(setup_p2p_test):
+    """
+    1. Create ls_db1 database on LiteServ One
+    2. Create ls_db2 database on LiteServ Two
+    3. Add 10000 docs to LiteServ 1 ls_db1
+    4. Create continuous push replication LiteServ 1 ls_db1 -> LiteServ 2 ls_db2
+    5. Verify all docs replicate to LiteServ 2 ls_db2
+    6. Verify all docs show up in changes for LiteServ 2 ls_db2
+    """
+
+    ls_url_one = setup_p2p_test["ls_url_one"]
+    ls_url_two = setup_p2p_test["ls_url_two"]
+
+    num_docs_per_db = 10000
+
+    log_info("ls_url_one: {}".format(ls_url_one))
+    log_info("ls_url_two: {}".format(ls_url_two))
+
+    client = MobileRestClient()
+
+    log_info("Creating databases")
+    ls_db1 = client.create_database(url=ls_url_one, name="ls_db1")
+    ls_db2 = client.create_database(url=ls_url_two, name="ls_db2")
+
+    ls_db1_docs = client.add_docs(url=ls_url_one, db=ls_db1, number=num_docs_per_db, id_prefix="test_ls_db1")
+
+    # Setup one shot push replication from LiteServ 1 ls_db1 to LiteServ 2 ls_db2
+    push_repl = client.start_replication(
+        url=ls_url_one,
+        continuous=False,
+        from_db=ls_db1,
+        to_url=ls_url_two, to_db=ls_db2,
+    )
+    log_info("Replication ID: {}".format(push_repl))
+
+    assert len(ls_db1_docs) == num_docs_per_db
 
     client.verify_docs_present(url=ls_url_two, db=ls_db2, expected_docs=ls_db1_docs)
     client.verify_docs_in_changes(url=ls_url_two, db=ls_db2, expected_docs=ls_db1_docs)
+
+
+@pytest.mark.sanity
+@pytest.mark.listener
+@pytest.mark.replication
+@pytest.mark.p2p
+@pytest.mark.changes
+def test_peer_2_peer_sanity_push_pull_one_shot(setup_p2p_test):
+    """
+    1. Create ls_db1 database on LiteServ One
+    2. Create ls_db2 database on LiteServ Two
+    3. Add 5000 docs to LiteServ 1 ls_db1
+    4. Add 5000 docs to LiteServ 2 ls_db2
+    5. Create continuous push replication LiteServ 1 ls_db1 -> LiteServ 2 ls_db2
+    6. Create continuous pull replication LiteServ 1 ls_db1 <- LiteServ 2 ls_db2
+    7. Verify all docs replicate to LiteServ 2 ls_db2
+    8. Verify all docs show up in changes for LiteServ 2 ls_db2
+    9. Verify all docs replicate to LiteServ 1 ls_db1
+    10. Verify all docs show up in changes for LiteServ 1 ls_db1
+    """
+
+    ls_url_one = setup_p2p_test["ls_url_one"]
+    ls_url_two = setup_p2p_test["ls_url_two"]
+
+    num_docs_per_db = 5000
+
+    log_info("ls_url_one: {}".format(ls_url_one))
+    log_info("ls_url_two: {}".format(ls_url_two))
+
+    client = MobileRestClient()
+
+    log_info("Creating databases")
+    ls_db1 = client.create_database(url=ls_url_one, name="ls_db1")
+    ls_db2 = client.create_database(url=ls_url_two, name="ls_db2")
+
+    ls_db1_docs = client.add_docs(url=ls_url_one, db=ls_db1, number=num_docs_per_db, id_prefix="test_ls_db1")
```

Comment:
Added the code but have commented it for now as addition through _bulk_docs needs more research